### PR TITLE
Introduction of gcc-11 to CI

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
Switch to gcc11 and g++11

### CODE REVIEW

**Comment by pwisniewskimobica** [comment]
We want to introduce compilation with gcc-11 in Flang CI.
This PR lets us build the gcc-11 artifacts for later use in Flang repository CI which is done in
flang-compiler/flang#1190

Kind request for your reviews, @bryanpkc , @shivaramaarao , @mleair , @kiranchandramohan
Thank you in advance!

**Comment by bryanpkc** [comment]
Please remove the change to `clang/test/Driver/Inputs/basic_cross_hurd_tree/usr/i386-gnu/bin/as`.

See my comment in #78.

**Comment by pwisniewskimobica** [comment]
> clang/test/Driver/Inputs/basic_cross_hurd_tree/usr/i386-gnu/bin/as

fixed

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by bryanpkc** [comment]
Since we are going to stop testing release_11x, I guess this PR can be closed.

**Comment by bryanpkc** [comment]
But if we decide to merge anyway, I would suggest rewording the commit to something like:
```
Use GCC 11 in CI builds
```

**Comment by pwisniewskimobica** [comment]
@bryanpkc commit message changed

**Review by RichBarton-Arm — APPROVED** [review]

**Comment by RichBarton-Arm** [comment]
@shivaramaarao , Huawei and Arm have approved. Are you happy also?

**Review by shivaramaarao — APPROVED** [review]
